src/System.Text.CaseFolding/System/Text/CaseFolding/SimpleCaseFolding.cs

```diff
@@ -0,0 +1,606 @@
+                        return compare1;
+                    }
+                }
+
+                if (length == 0)
+                {
+                    return result;
+                }
+
+                while (length != 0 && !IsSurrogate(refA) && !IsSurrogate(refB))
+                {
+                    var compare2 = SimpleCaseFoldCompareAbove05ff(refA, refB, ref refMapLevel1, ref refMapData);
+
+                    if (compare2 == 0)
+                    {
+                        length--;
+                        refA = ref Unsafe.Add(ref refA, 1);
+                        refB = ref Unsafe.Add(ref refB, 1);
+                    }
+                    else
+                    {
+                        return compare2;
+                    }
+                }
+
+                if (length == 0)
+                {
+                    return result;
+                }
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Simple case folding of the string.
+        /// </summary>
+        /// <param name="source">Source string.</param>
+        /// <returns>
+        /// Returns new allocated folded string.
+        /// </returns>
+        //[MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static string SimpleCaseFold(this string source)
+        {
+            if (source == null || source.Length == 0)
+            {
+                return source;
+            }
+
+            return string.Create(
+                source.Length,
+                source,
+                (chars, sourceString) =>
+                {
+                    SpanSimpleCaseFold(chars, sourceString);
+                });
+        }
+
+        /// <summary>
+        /// Simple case folding of the Span&lt;char&gt; on place.
+        /// </summary>
+        /// <param name="source">Source and target span.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void SimpleCaseFold(this Span<char> source)
+        {
+            SpanSimpleCaseFold(source, source);
+        }
+
+        /// <summary>
+        /// Simple case folding of the ReadOnlySpan&lt;char&gt;.
+        /// </summary>
+        /// <param name="source">Source span.</param>
+        /// <returns>
+        /// Returns new allocated folded span.
+        /// </returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Span<char> SimpleCaseFold(this ReadOnlySpan<char> source)
+        {
+            Span<char> destination = new char[source.Length];
+
+            SpanSimpleCaseFold(destination, source);
+
+            return destination;
+        }
+
+        /// <summary>
+        /// Simple case folding of the ReadOnlySpan&lt;char&gt;.
+        /// </summary>
+        /// <param name="source">Destination span.</param>
+        /// <param name="source">Source span.</param>
+        public static void SpanSimpleCaseFold(Span<char> destination, ReadOnlySpan<char> source)
```

Comment:
>SpanSimpleCaseFold [](start = 27, length = 18)

I would remove the word Span from the name. the parameters is telling we are working on spans

Reply:
Renamed SpanSimpleCaseFold() to SimpleCaseFold().